### PR TITLE
Create CompileSassView

### PR DESCRIPTION
**Pull request by estherjsuh**
## Create `CompileSassView`

This api compiles microsite's SASS by calling `SiteConfiguration.compile_microsite_sass()`

```
POST /appsembler/api/compile_sass/
   {"site_uuid": "fake-site-uuid"}
```

### CODE REVIEW

File: openedx/core/djangoapps/appsembler/sites/api.py [hunk 1]
```diff
@@ -270,3 +270,13 @@ def post(self, request, format=None):
 class CustomDomainView(CreateAPIView):
     queryset = AlternativeDomain.objects.all()
     serializer_class = AlternativeDomainSerializer
+
+
+class CompileSassView(APIView):
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/api.py:275]:
```suggestion
class CompileSassView(APIView):
    """
    TODO: Document this API.
    """

    permission_classes = [ApiKeyHeaderPermission]

```

File: openedx/core/djangoapps/appsembler/sites/api.py [hunk 2]
```diff
@@ -270,3 +270,13 @@ def post(self, request, format=None):
...
+    def post(self, request, format=None):
+        site_uuid = request.data.get('site_uuid')
+        org = Organization.objects.get(edx_uuid=site_uuid)
+        site = org.sites.get()
+        configuration = SiteConfiguration.objects.get(site=site)
+        configuration.compile_microsite_sass()
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/sites/api.py:281]: This is needed so `SiteConfiguration` model uses the service adapter.

```suggestion
        configuration.init_api_client_adapter(site)
        configuration.compile_microsite_sass()
```

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Thanks @estherjsuh. I've added few changes for the API view.

**Comment by estherjsuh** [comment]
@OmarIthawi I appreciate the review. I made the suggested changes and added testing - please give this PR another pass. Thanks!

**Review by OmarIthawi — APPROVED** [review]

**Comment by OmarIthawi** [comment]
Thanks @estherjsuh! Looks great now, please deploy for testing on staging.

**Review by thraxil — APPROVED** [review]

**Pull request merged**